### ISSUE TITLE
[SQA]-THREAD TC-DL-2.14 Unable to execute the credential commands.

### ISSUE THREAD

**Issue by Survensa** [project-chip/connectedhomeip#18418]
**Problem:**
In TC-DL-2.14 The Credential command(set,get,clear) are unable to execute in the Chip-tool as we are unable to provide Struct values for Credentials (Credential Type and Credential Index)

**Excepted outcome** :  Success response

**Actual Outcome** : Unable to execute the Command

**Step to reproduce:**  `./chip-tool doorlock set-credential 0 1 2 xxxx 2 0 0 1 1`

**Platform :** RPI - RPI

**Commit** used: 11587f5bf0bb9e621edd4c867c0bc942d804d1ce

**Reference:**  [Test-Plan](https://github.com/CHIP-Specifications/chip-test-plans/blob/master/src/cluster/door_lock_Cluster.adoc)

**Logs:**

> ./chip-tool doorlock set-credential OperationType Credential CredentialData UserIndex UserStatus UserType node-id/group-id endpoint-id-ignored-for-group-commands [--paa-trust-store-path] [--commissioner-name] [--commissioner-nodeid] [--trace_file] [--trace_log] [--ble-adapter] [--timedInteractionTimeoutMs] [--suppressResponse] [--repeat-count] [--repeat-delay-ms] [--timeout]
[1652348232.057935][30233:30233] CHIP:TOO: Run command failure: ../../commands/common/Commands.cpp:137: Error 0x0000002F

**Comment by jrhees-cae** [comment]
The format of the command is incorrect. The Credential parameter needs to be JSON-encoded. For example:
`chip-tool doorlock set-credential 0 '{"credentialType":1, "credentialIndex":1}' '5555' 1 1 0 555 1 --timedInteractionTimeoutMs 1000`

See https://github.com/CHIP-Specifications/chip-test-plans/issues/1298

**Comment by cjandhyala** [comment]
@Survensa Can we close this ticket ?

**Comment by woody-apple** [comment]
Cert Blocker Review: Assigning to myself and @franck-apple to find an assignee. Tagging @bzbarsky-apple as well

**Comment by bzbarsky-apple** [comment]
@Survensa Please attach (not paste) complete logs from the client (including, critically, the exact command being executed) and the server.

My guess is that this is using a credential type not supported by the server app, but the logs would tell for sure.

**Comment by Survensa** [comment]
@bzbarsky-apple Here I have attached the log of server and client for this command .
[setcerdential-cntl.txt](https://github.com/project-chip/connectedhomeip/files/8778021/setcerdential-cntl.TXT)
[setcerdential-dut.txt](https://github.com/project-chip/connectedhomeip/files/8778022/setcerdential-dut.txt)

**Comment by cjandhyala** [comment]
@jrhees-cae this test is run on nordic nrfdk board

**Comment by bzbarsky-apple** [comment]
Yep, that matches my guess from https://github.com/project-chip/connectedhomeip/issues/18418#issuecomment-1137600252

@cjandhyala @Survensa does the feature map for this device claim PIN support?

**Comment by sethunk** [comment]
@bzbarsky-apple In lock app, Feature map value: 275 
Binary value : 100010011 ( Which supports PIN,RID,SCH,NOT )

**Comment by jrhees-cae** [comment]
FeatureMap value for door-lock example is likely common to all platforms implemented for the example. https://github.com/project-chip/connectedhomeip/blob/master/examples/lock-app/lock-common/lock-app.zap

Therefore, if a certain platform example doesn't support the common features, it would need to override the featuremap value.

**Comment by franck-apple** [comment]
Pinging @kkasperczyk-no and @Damian-Nordic

**Comment by bzbarsky-apple** [comment]
Yes, sounds like some apps need to set the value at startup, not just use the default from the ZAP file...

**Comment by kkasperczyk-no** [comment]
@Survensa @cjandhyala can we close that one? The only issue I see are differences between feature maps vs the real features that are implemented, but it affects probably more examples. If we want to fix this, it could be addressed in a more generic issue that this one.

**Comment by cjandhyala** [comment]
per @kkasperczyk-no nordic platform chose not to implement a USR feature, hence this is not a cert blocker

**Comment by bzbarsky-apple** [comment]
@cjandhyala Note that the USR feature is required if any of PIN, RID, FPG, or FACE are supported.

**Comment by kkasperczyk-no** [comment]
@bzbarsky-apple right, however nRF platform in particular doesn't implement any of those.
